src/server.h

```diff
@@ -1187,16 +1188,17 @@ typedef struct client {
     uint64_t id; /* Client incremental unique ID. */
     connection *conn;
     /* Input buffer and command parsing fields */
-    sds querybuf;        /* Buffer we use to accumulate client queries. */
-    size_t qb_pos;       /* The position we have read in querybuf. */
-    robj **argv;         /* Arguments of current command. */
-    int argc;            /* Num of arguments of current command. */
-    int argv_len;        /* Size of argv array (may be more than argc) */
-    size_t argv_len_sum; /* Sum of lengths of objects in argv list. */
-    int reqtype;         /* Request protocol type: PROTO_REQ_* */
-    int multibulklen;    /* Number of multi bulk arguments left to read. */
-    long bulklen;        /* Length of bulk argument in multi bulk request. */
-    long long woff;      /* Last write global replication offset. */
+    sds querybuf;         /* Buffer we use to accumulate client queries. */
+    size_t qb_pos;        /* The position we have read in querybuf. */
+    bool is_qb_full_read; /* True if the last read returned the maximum allowed bytes */
```

Comment:
This adds 4 bytes to every client in primaries and replicas, but we only need it for one very specific client.

Some other ideas to save memory:

* Use a flag. I know we have very few unused flags left, but it's fine to use them. (When it's full, we can add another 32 or 64 bits for more flags.)
* Change the return type of `readToQueryBuf` from void to bool or an enum, to indicate if it was a full or a short read.